models/Challenge.js

```diff
@@ -24,156 +24,55 @@ class Challenge {
   }
 
   static create(userId, spotId, performanceId) {
-    const sql = 'SELECT make_challenge($1, $2, $3)';
-
     return new Promise((resolve, reject) => {
-      const client = utils.db.createClient();
-      let returnCode;
-
-      client.connect();
-      client.on('error', (err) => reject(err));
-
-      const query = client.query(sql, [userId, performanceId, spotId]);
-
-      query.on('row', ({ make_challenge }) => {
-        returnCode = parseInt(make_challenge, 10);
-      });
+      const sql = 'SELECT make_challenge($1, $2, $3)';
 
-      query.on('end', () => {
-        client.end();
-        resolve(returnCode);
-      });
-
-      query.on('error', (err) => {
-        client.end();
-        reject(err);
-      });
+      db.query(sql, [userId, performanceId, spotId], codeFromRow)
+      .then(resolve)
+      .catch(reject);
     });
   }
 
   static findAllChallengeablePeopleForUser(user, performanceId) {
-    // eslint-disable-next-line
-    const sql = queries.challengeablePeople;
-
     return new Promise((resolve, reject) => {
       if (!performanceId) {
         resolve(null);
       }
+      const sql = queries.challengeablePeople;
 
-      const client = utils.db.createClient();
-      const users = [];
-
-      client.connect();
-      client.on('error', (err) => reject(err));
-
-      const query = client.query(sql, [user.instrument, user.part, user.spotId, user.nameNumber, performanceId]);
-
-      query.on('row', ({ challengedcount, name, spotid, open }) => {
-        users.push(new ChallengeableUser(challengedcount, name, spotid, open));
-      });
-
-      query.on('end', () => {
-        client.end();
-        resolve(users);
-      });
-
-      query.on('error', (err) => reject(err));
+      db.query(sql, [user.instrument, user.part, user.spotId, user.nameNumber, performanceId], instanceFromRowChallengeableUser)
+      .then(resolve)
```

Comment:
Instead of wrapping this in return new promise, just return the promise directly, no then resolve catch reject. Above instead of resolve null, return Promisw.resolve(null)